Language/Haskell/GhcMod/Logging.hs

```diff
@@ -78,6 +83,17 @@ gmLog level loc' doc = do
 
   gmlJournal (GhcModLog Nothing (Last Nothing) [(level, loc', msgDoc)])
 
+-- | Appends a collection of logs to the logging environment, with effects
+-- | if their log level specifies it should
+gmLog' :: (MonadIO m, GmLog m, GmOut m) => GhcModLog -> m ()
+gmLog' GhcModLog { gmLogMessages } = do
```

Comment:
Why not just ```gmLog `mapM` gmLogMessages``` (plus-minus some argument shuffling)? That takes care of all the filtering and whatnot and you don't need to convert anything to a string.